### PR TITLE
Delete proxied state from the dagster process

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

Move the proxied state stuff to `in_airflow` and delete all code related to it in the dagster process.

## How I Tested These Changes

BK

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by schrockn** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25007?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25007** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25007?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24981** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24981?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @schrockn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py [hunk 1]
```diff
@@ -85,19 +78,6 @@ def list_variables(self) -> List[Dict[str, Any]]:
                 "Failed to fetch variables. Status code: {response.status_code}, Message: {response.text}"
             )
 
-    def get_proxied_state(self) -> AirflowProxiedState:
-        local_migration_dir = get_local_proxied_state_dir()
-        if local_migration_dir is not None:
-            return load_proxied_state_from_yaml(local_migration_dir)
-        variables = self.list_variables()
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py:92]: I'm keeping list_variables around in case we need it again.

Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py:92]: seems reasonable

**Review by dpeng817 — APPROVED** [review]

**Comment by dpeng817** [comment]
This looks good. As a followup we should also remove all variable munging.

**Comment by schrockn** [comment]
> This looks good. As a followup we should also remove all variable munging.

which munging, specifically?

**Comment by dpeng817** [comment]
@schrockn all the stuff we do in the in_airflow codebase to set Variable state, and in the examples all the local file stuff we do.

**Comment by dpeng817** [comment]
Landing this to unblock Proxy Operator stack

**Pull request merged**